singer_sdk/typing.py

```diff
@@ -352,21 +352,30 @@ def __init__(
         required: bool = False,
         default: _JsonValue = None,
         description: str = None,
+        secret: bool = False,
     ) -> None:
         """Initialize Property object.
 
+        Note: Properties containing secrets should be specified with `secret=True`.
+        Doing so will add the annotation `writeOnly=True`, in accordance with JSON
+        Schema Draft 7 and later, and `secret=True` as an additional hint to readers.
+
+        More info: https://json-schema.org/draft-07/json-schema-release-notes.html
```

Comment:
@aaronsteers Should we update the validator in `singer_sdk/plugin_base.py` to use the `Draft7Validator` object?

Reply:
@edgarrmondragon - I don't know, honestly. We don't rely on the validator taking any action from these hints, so I thin it is fine to keep things as they are. Optionally, we can bump to Draft7 and confirm that nothing breaks. I don't have a strong feeling either way.

Reply:
We'll see! 

👀 https://github.com/meltano/sdk/pull/1096/commits/d9647bd412db8db7b2c91aeb49b7955576ed34b5

Reply:
Looks like no issues. 👍